### PR TITLE
Feat/android chartbeat

### PR DESCRIPTION
**Pull request by drazumeev-sc**
## Initial implementation of Chartbeat analytics for Android

[Monday ticket](https://applicaster.monday.com/boards/1615228456/pulses/2796268626)

Initial implementation of Chartbeat analytics for Android

<details>
  <summary>Platforms Affected</summary>
  
- [x] Android
- [x] Android TV
- [ ] Apple TV (TVOS)
- [ ] Fire TV (FireOS)
- [ ] iOS
- [ ] LG TV (WebOS)
- [ ] Samsung TV (Tizen)
- [ ] Web
</details>

## Known or Pending Issues

`fill in this section with potential issues / limitations the reviewer(s) should know about`

## Screenshots

`If there is some kind of ui change please share screenshots`

## Version Info

`Does this update require a certain version of quickbrick, or does it rely on another pr to work?`

### CODE REVIEW

File: plugins/zapp-analytics-plugin-chartbeat/android/src/main/java/com/applicaster/chartbeat/ChartbeatAnalyticsAgent.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    override fun initializeAnalyticsAgent(context: Context?) {
+        super.initializeAnalyticsAgent(context)
+
+        routers.clear()
+        routers.add(ScreenAdapter())
+    }
+
+    override fun setParams(params: MutableMap<Any?, Any?>?) {
+        super.setParams(params)
+        val pluginConfig = PluginConfigParser().getConfig(params)
```
Comment (inckie) [inline plugins/zapp-analytics-plugin-chartbeat/android/src/main/java/com/applicaster/chartbeat/ChartbeatAnalyticsAgent.kt:26]: Lets validate config and log error if we missing required fields, so CS will know what broken before they came to us

File: plugins/zapp-analytics-plugin-chartbeat/manifests/manifest.config.js [hunk 2]
```diff
@@ -95,14 +129,24 @@ const api_apple = {
   modules: ["ZappAnalyticsPluginChartbeat"]
 };
 
+const api_android = {
+  require_startup_execution: false,
+  class_name: "com.applicaster.chartbeat.ChartbeatAnalyticsAgent"
```
Comment (inckie) [inline plugins/zapp-analytics-plugin-chartbeat/manifests/manifest.config.js:134]: lets change package to com.applicaster.analytics
I do it for new analytics plugins for consistency, like com.applicaster.analytics.adobe, com.applicaster.analytics.airship, com.applicaster.analytics.gemius, com.applicaster.analytics.comscore

File: plugins/zapp-analytics-plugin-chartbeat/android/src/main/java/com/applicaster/chartbeat/ChartbeatAnalyticsAgent.kt (hunk 1 above)
Comment (drazumeev-sc) [reply plugins/zapp-analytics-plugin-chartbeat/android/src/main/java/com/applicaster/chartbeat/ChartbeatAnalyticsAgent.kt]: Agree, done

File: plugins/zapp-analytics-plugin-chartbeat/manifests/manifest.config.js (hunk 2 above)
Comment (drazumeev-sc) [reply plugins/zapp-analytics-plugin-chartbeat/manifests/manifest.config.js]: Done

**Pull request merged**